numbthy.py

```diff
@@ -235,7 +236,7 @@ def factors(n):
 def factorPR(n):
 	"""factorPR(n) - Find a factor of n using the Pollard Rho method.
 	Note: This method will occasionally fail."""
-	numsteps=2*math.floor(math.sqrt(math.sqrt(n)))
+	numsteps=2*math.floor(decimal.Decimal((decimal.Decimal(n).sqrt()).sqrt()))
```

Comment:
The proposed change is, on the face of it, correct, but this is a place in the algorithm where accuracy is not needed.  The formula for numsteps is an asymptotic result and, as is the case for such things, is not itself accurate for ranges we are working with (say n < 2^2048).  Thus, improving the accuracy when computing the formula is not worth adding a new library as a dependency.  (See wikipedia page for Pollard Rho or Sect 14.2 of Mathematics of Public Key Cryptography, Version 2.0 by S. D. Galbraith [https://www.math.auckland.ac.nz/~sgal018/crypto-book/main.pdf])